fittings/swagger_router.js

```diff
@@ -47,56 +48,86 @@ module.exports = function create(fittingDef, bagpipes) {
+        debug('created mock controller function %s.%s', controllerName, operationId);
+      }
+
       if (controllerFunction && typeof controllerFunction === 'function') {
-        debug('running controller');
+        debug('running controller %s.%s', controllerName, operationId);
         return controllerFunction(context.request, context.response, cb);
       }
 
-      var msg = util.format('Controller %s doesn\'t export handler function %s', controllerName, operationId);
-      if (mockMode) {
-        debug(msg);
-      } else {
-        return cb(new Error(msg));
-      }
+      debug(util.format('Controller %s doesn\'t export handler function %s', controllerName, operationId));
+      return cb(new Error(msg));
     }
 
-    if (mockMode) {
+    // for completeness, we should never actually get here
+    cb(new Error(util.format('No controller found for %s in %j', controllerName, controllersDirs)));
+  }
+};
+
+function createMockControllerFunction(fittingDef, bagpipes) {
+  var swaggerNodeRunner = bagpipes.config.swaggerNodeRunner;
+  var appRoot = swaggerNodeRunner.config.swagger.appRoot;
+
+  var mockModeNoCallback = !!fittingDef.mockModeNoCallback || !!swaggerNodeRunner.config.swagger.mockModeNoCallback;
+
+  return function default_mock_controller(request, response, cb) {
+    debug('exec default_mock_controller')
 
-      var statusCode = parseInt(context.request.get('_mockreturnstatus')) || 200;
+    var operation = request.swagger.operation;
+    var controllerName = operation[SWAGGER_ROUTER_CONTROLLER] || operation.pathObject[SWAGGER_ROUTER_CONTROLLER];
 
-      var mimetype = context.request.get('accept') || 'application/json';
-      var mock = operation.getResponseExample(statusCode, mimetype);
+    var statusCode = parseInt(request.get('_mockreturnstatus')) || 200;
 
-      if (mock) {
-        debug('returning mock example value', mock);
+    var mimetype = request.get('accept');
+    if (!mimetype || mimetype === '*/*') {
+      if (!operation.produces || !operation.produces.length || operation.produces.indexOf('application/json') >= 0) {
+        mimetype = 'application/json';
       } else {
-        mock = operation.getResponseSample(statusCode);
-        debug('returning mock sample value', mock);
+        mimetype = operation.produces[0];
       }
+    }
 
-      context.headers['Content-Type'] = mimetype;
-      context.statusCode = statusCode;
+    var mock = operation.getResponseExample(statusCode, mimetype);
 
-      return cb(null, mock);
+    if (mock) {
+      debug('returning mock example value', mock);
+    } else {
+      mock = operation.getResponseSample(statusCode);
+      debug('returning mock sample value', mock);
     }
 
-    // for completeness, we should never actually get here
-    cb(new Error(util.format('No controller found for %s in %j', controllerName, controllersDirs)));
-  }
-};
+    response.setHeader('Content-Type', mimetype);
```

Comment:
Can someone please confirm if it's ok to use response.setHeader instead of context.headers[]?